ServerCore/PuzzleStateHelper.cs

```diff
@@ -168,6 +168,25 @@ public static async Task SetSolveStateAsync(PuzzleServerContext context, Event e
                 states[i].SolvedTime = value;
             }
 
+            // Award hint coins
+            if (value != null)
```

Comment:
"&& puzzle.HintCoinsForSolve != 0", for perf

Reply:
Good idea. Done.